### PR TITLE
Add error messages and line numbers for overflowing numeric operations

### PR DESCRIPTION
**Pull request by elliotkennedy**

### CODE REVIEW

File: programs/hyperplane/src/utils/macros.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ macro_rules! curve {
 macro_rules! require_msg {
     ($invariant:expr, $error:expr $(,)?, $message: expr) => {
         if !($invariant) {
-            msg!($message);
+            anchor_lang::prelude::msg!($message);
```
Comment (oeble) [inline programs/hyperplane/src/utils/macros.rs:44]: If you want the "right" way to use an external crate feature in a macro:
```suggestion
            ::anchor_lang::prelude::msg!($message);
```

File: programs/hyperplane/src/utils/macros.rs [hunk 2]
```diff
@@ -99,3 +99,25 @@ macro_rules! to_u64 {
...
+/// Macro to wrap a math operation in a result with useful error message and line number
+#[macro_export]
+macro_rules! optional_math {
+    ($val: expr) => {
+        $val.ok_or_else(|| {
+            anchor_lang::prelude::msg!("[{}:{}] {}", file!(), line!(), stringify!($val));
+            anchor_lang::error!($crate::error::SwapError::CalculationFailure)
+        })
+    };
+}
```
Comment (oeble) [inline programs/hyperplane/src/utils/macros.rs:123]: If you replace all your math with `try_math` in which case this is still useful?

**Review by oeble — APPROVED** [review]
Very interesting idea. Maybe you might also want to give a look at how [`checked_expr`](https://docs.rs/checked_expr/0.3.0/checked_expr/macro.checked_expr.html) work under the hood and makes something similar that will use `try_math` under the hood so restore normal mathematical notation in our codebases (A dream of mine)

File: programs/hyperplane/src/utils/macros.rs (hunk 2 above)
Comment (elliotkennedy) [reply programs/hyperplane/src/utils/macros.rs:123]: It is for the time-being - I haven't migrated every curve method to return Result yet - so using it for those

File: programs/hyperplane/src/utils/macros.rs (hunk 1 above)
Comment (elliotkennedy) [reply programs/hyperplane/src/utils/macros.rs:44]: Neat thanks!

**Comment by elliotkennedy** [comment]
> Very interesting idea. Maybe you might also want to give a look at how [`checked_expr`](https://docs.rs/checked_expr/0.3.0/checked_expr/macro.checked_expr.html) work under the hood and makes something similar that will use `try_math` under the hood so restore normal mathematical notation in our codebases (A dream of mine)

This is interesting - will have a look thanks

**Comment by elliotkennedy** [comment]
> > Very interesting idea. Maybe you might also want to give a look at how [`checked_expr`](https://docs.rs/checked_expr/0.3.0/checked_expr/macro.checked_expr.html) work under the hood and makes something similar that will use `try_math` under the hood so restore normal mathematical notation in our codebases (A dream of mine)
> 
> This is interesting - will have a look thanks

I was thinking about adding `overflow-checks = true` earlier but it doesn't feel solid enough for some reason

**Comment by oeble** [comment]
> > > Very interesting idea. Maybe you might also want to give a look at how [`checked_expr`](https://docs.rs/checked_expr/0.3.0/checked_expr/macro.checked_expr.html) work under the hood and makes something similar that will use `try_math` under the hood so restore normal mathematical notation in our codebases (A dream of mine)
> > 
> > 
> > This is interesting - will have a look thanks
> 
> I was thinking about adding `overflow-checks = true` earlier but it doesn't feel solid enough for some reason

It's solid but panic instead of error which is not great for the user experience (not a proper message)

**Pull request merged**